### PR TITLE
Add Dojo 2 as a Preset Template fixes #447

### LINKED ISSUE #447: Dojo 2 Support
It would be awesome if we could get support for a Dojo 2 template.

Dojo 2 uses Webpack for bundling via the [dojo cli](https://github.com/dojo/cli-build-app). Please let us know if there is anything we can do to help with this!

### PR DESCRIPTION
**Pull request by nicknisi**
**What kind of change does this PR introduce?**

This PR adds a new template for Dojo 2.

**What is the current behavior?**
This is a new template, requested and described in #447 

**What is the new behavior?**

Dojo 2 template

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged 
- [ ] Added myself to contributors table 

- add dojo2 template
- add temporary project as template to load (github)
- add type definition creation to styles transpiler
- add dojo 2 to home page
- add default loading of dojo template src/index.html and src/main.css

**Note:** This has a backend change in order to set the correct template value coming from the API. To work around this in development, I added the following line to packages/app/src/app/store/providers/Api.js:

```diff
diff --git i/packages/app/src/app/store/providers/Api.js w/packages/app/src/app/store/providers/Api.js
index 35248ccb..db410a04 100644
--- i/packages/app/src/app/store/providers/Api.js
+++ w/packages/app/src/app/store/providers/Api.js
@@ -78,6 +78,7 @@ function handleResponse(response, { shouldCamelize = true } = {}) {
     camelizedData.data.npmDependencies = response.result.data.npm_dependencies;
   }
 
+  camelizedData.data.template = '@dojo/cli-create-app';
   return camelizedData.data ? camelizedData.data : camelizedData;
 }
```

### CODE REVIEW

File: packages/app/src/app/components/CodeEditor/Monaco/index.js [hunk 1]
```diff
@@ -199,6 +202,18 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
     }
   }
 
+  setupTranspilationListener() {
+    listen(({ type, code, path }) => {
```
Comment (CompuIves) [inline packages/app/src/app/components/CodeEditor/Monaco/index.js]: This is reaaaally nice! I love that we now have the power of templates to define type definitions.

File: packages/app/src/sandbox/eval/transpilers/style/index.js [hunk 2]
```diff
@@ -24,12 +29,14 @@ class StyleTranspiler extends Transpiler {
...
   doTranspilation(code: string, loaderContext: LoaderContext) {
     const id = getStyleId(loaderContext._module.getId());
+    const { path } = loaderContext;
 
     if (loaderContext.options.module) {
       return getModules(code, loaderContext).then(({ css, exportTokens }) => {
         let result = insertCss(id, css);
         result += `\nmodule.exports=${JSON.stringify(exportTokens)};`;
 
+        dispatch({ type: 'add-extra-lib', path, code: classesToDefinition(exportTokens) });
```
Comment (CompuIves) [inline packages/app/src/sandbox/eval/transpilers/style/index.js]: I LOVE THIS EVEN MORE!

**Comment by nicknisi** [comment]
@CompuIves Wow, thanks for the feedback! I went ahead and pushed up a commit with the API override for now so you should be able to test on staging.

I would be happy to document my learning and some of the difficulties! I'll start writing down my thoughts after work tonight.

**Comment by CompuIves** [comment]
I made some changes in `master`, I'm sorry for that! I changed how Icons are resolved to utilize better code splitting. Could you rebase your branch on master? I'm planning to merge this in next Thursday.

**Comment by nicknisi** [comment]
@CompuIves I rebased off the latest `master` and removed `DojoIcon` from the dojo template. That seemed to be all I needed to do to get it to work again! Let me know if there's any other way I can help!

Also, I am still working on a write-up about adding a new template. I should have it by this weekend at the latest.

**Comment by dylans** [comment]
@CompuIves any update on when this might land? We're very close to releasing Dojo 2 and would like to mention this in the blog post announcing the Dojo 2 release. Thanks!

**Comment by CompuIves** [comment]
@dylans I will have more time for CodeSandbox from tomorrow. So I'm either planning on releasing it today or tomorrow, sorry for the delay!

**Comment by CompuIves** [comment]
Whoop! It's in now!

**Pull request closed without merging**